Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -592,13 +592,16 @@ void NetworkProcess::destroySession(PAL::SessionID sessionID)
         session->invalidateAndCancel();
         auto& storageManager = session->storageManager();
         m_closingStorageManagers.add(&storageManager);
-        storageManager.close([this, protectedThis = Ref { *this }, storageManager = &storageManager]() {
+        storageManager.close([this, protectedThis = Ref { *this }, storageManager = &storageManager, completionHandler = std::exchange(completionHandler, { })]() mutable {
             m_closingStorageManagers.remove(storageManager);
+            completionHandler();
```

Comment:
It would be good to have some API testing. I am a little worried the IPC back won't reliably make it since we're doing this right before exiting the process.

Reply:
Added API test. 
If network process exits, won't UI process error out the pending message reply?

Reply:
Yes, I think that's right. Even though calling the completionHandler on the network process side may not work because we're about to exit. The UIProcess would call the completion handler on its side if there is a pending sendWithAsyncReply() when the IPC connection gets severed.